### PR TITLE
add menu contribution for python manage config

### PR DESCRIPTION
**Pull request by eleanorjboyd**
related to this issue: https://github.com/microsoft/vscode-python/issues/21845

### CODE REVIEW

File: src/vs/workbench/services/actions/common/menusExtensionPoint.ts [hunk 1]
```diff
@@ -343,6 +343,11 @@ const apiMenus: IAPIMenu[] = [
 		id: MenuId.TestItemGutter,
 		description: localize('testing.item.gutter.title', "The menu for a gutter decoration for a test item"),
 	},
+	{
+		key: 'testing/pythonConfig',
```
Comment (connor4312) [inline src/vs/workbench/services/actions/common/menusExtensionPoint.ts:347]: This isn't just for Python, other extensions can use it as well. It should be named more generically.

Perhaps `testing/profiles/context`

File: src/vs/workbench/contrib/testing/browser/testingExplorerView.ts [hunk 2]
```diff
@@ -351,9 +352,19 @@ export class TestingExplorerView extends ViewPane {
 			}
 		}
 
-		// If there's only one group, don't add a heading for it in the dropdown.
-		if (participatingGroups === 1) {
-			profileActions.shift();
+		const menuActions: IAction[] = [];
+
+		const key = this.contextKeyService.createOverlay([]);
```
Comment (connor4312) [inline src/vs/workbench/contrib/testing/browser/testingExplorerView.ts:357]: One key we probably should have in the overlap is the current group (run/debug/coverage) for extensions who want different behavior

File: src/vs/workbench/contrib/testing/browser/testingExplorerView.ts [hunk 3]
```diff
@@ -351,9 +352,19 @@ export class TestingExplorerView extends ViewPane {
...
+		const menu = this.menuService.createMenu(MenuId.TestPythonConfigMenu, key);
+		const actions = menu.getActions({ renderShortTitle: true }).flatMap(entry => entry[1]);
+		if (actions.length > 0) {
+			// fill if there are any actions
```
Comment (connor4312) [inline src/vs/workbench/contrib/testing/browser/testingExplorerView.ts:361]: are these lines necessary? `getActions` just gets the actions that it'll fill in when you call `createAndFillIn` anyway

File: src/vs/workbench/contrib/testing/browser/testingExplorerView.ts (hunk 2 above)
Comment (eleanorjboyd) [reply src/vs/workbench/contrib/testing/browser/testingExplorerView.ts:357]: Hi, sorry could you clarify a bit what you mean here? There should be a context key for the current group and have it only show for some or make that configurable for the extension author?

Comment (connor4312) [reply src/vs/workbench/contrib/testing/browser/testingExplorerView.ts:357]: The menu logic can run for all of them, but an author might want to only have a menu visible in one group, or have different menu entries for each group. So letting them do something like `profileKind == "run"` would be good.

File: src/vs/workbench/contrib/testing/browser/testingExplorerView.ts [hunk 4]
```diff
@@ -351,10 +352,29 @@ export class TestingExplorerView extends ViewPane {
...
+		// allow extension author to define context for when to show the test menu actions for run or debug menus
+		if (group === TestRunProfileBitset.Run) {
+			contextKeys.push(['testing.profile.context.group', 'run']);
+		}
+		if (group === TestRunProfileBitset.Debug) {
+			contextKeys.push(['testing.profile.context.group', 'debug']);
+		}
+		if (group === TestRunProfileBitset.Coverage) {
+			contextKeys.push(['testing.profile.context.group', 'coverage']);
 		}
```
Comment (connor4312) [inline src/vs/workbench/contrib/testing/browser/testingExplorerView.ts:366]: Can you register the key in testingContextKeys.ts? That way it'll be automatically picked up for our docs :)

**Review by connor4312 — COMMENTED** [review]
Looking good, one comment

File: src/vs/workbench/contrib/testing/browser/testingExplorerView.ts (hunk 4 above)
Comment (eleanorjboyd) [reply src/vs/workbench/contrib/testing/browser/testingExplorerView.ts:366]: yes!

**Review by justschen — APPROVED** [review]

**Pull request merged**